src/saltext/vmware/utils/connect.py

```diff
@@ -78,26 +77,38 @@ def get_service_instance(opts=None, pillar=None, esxi_host=None):
                     user: admin
                     password: ***
 
+    If configuration for multiple VMWare services instances is required, they can be
+    set up as different configuration profiles:
+    For example:
+    .. code-block:: yaml
+
+        vmware_config:
+            profile1:
+                host: 198.51.100.100
+                password: ****
+                user: @example.com
+            profile2:
+                host: 198.51.100.100
+                password: ****
+                user: @example.com
+                esxi_host:
+                    198.52.100.105:
+                        user: admin
+                        password: ***
+                    198.52.100.106:
+                        user: admin
+                        password: ***
+
     """
     ctx = ssl._create_unverified_context()
-    opts = opts or {}
-    pillar = pillar or {}
-    host = (
-        esxi_host
-        or os.environ.get("VMWARE_CONFIG_HOST")
-        or opts.get("vmware_config", {}).get("host")
-        or pillar.get("vmware_config", {}).get("host")
-    )
-    user, password = get_username_password(esxi_host=host, opts=opts, pillar=pillar)
-    config = {
-        "host": host,
-        "password": password,
-        "user": user,
-    }
+    config = config or {}
+
+    config = get_config(config=config, profile=profile, esxi_host=esxi_host)
+
     service_instance = connect.SmartConnect(
-        host=config["host"],
-        user=config["user"],
-        pwd=config["password"],
+        host=config[0],
+        user=config[1],
+        pwd=config[2],
```

Comment:
**question** can we not use key access here? Index access is easier to break without things really breaking :thinking:

Reply:
Done